curvlinops/kfac.py

```diff
@@ -279,6 +280,23 @@ def draw_label(self, output: Tensor) -> Tensor:
                 generator=self._generator,
             )
             return output.clone().detach() + perturbation
+
+        elif isinstance(self._loss_func, CrossEntropyLoss):
+            # TODO For output.ndim > 2, the scale of the 'would-be' gradient resulting
+            # from these labels might be off
+            if output.ndim != 2:
+                raise NotImplementedError(
+                    "Only 2D output is supported for CrossEntropyLoss for now."
+                )
+            probs = output.softmax(dim=1)
+            # each row contains a vector describing a categorical
+            probs_as_mat = rearrange(probs, "n c ... -> (n ...) c")
```

Comment:
The `softmax` and `rearrange` operation could just be executed once before the for-loop over `range(self._mc_samples)` in a `_preprocess_model_output(output)` function or so.

Reply:
Hm. Not sure if this will be relevant in practise, as I believe `mc_samples` is almost always `1`, at least for the use cases I know.